src/stdLambda.cls

```diff
@@ -747,11 +764,143 @@ Private Sub parseFlowPriority1()
         
             Call optConsume("end")
         End If
+    Else
+        Call parseFlowPriority2
+    End If
+End Sub
+
+'Parse loops (do ... until|while ... loop)
+Private Sub parseFlowPriority2()
+    If optConsume("do") Then
+        Dim RepeatLoopIndex As Integer: RepeatLoopIndex = This.iOperationIndex
+        Dim RepeatLoopJumpIndex As Integer
+        Dim ExitLoopJumpIndex As Integer: ExitLoopJumpIndex = -1 ' -1 to ensure a value, which means "no exit"
+        If optConsume("until") Then
+            Call parseExpression                                     ' Break Condition
+            RepeatLoopJumpIndex = addOperation(oJump, ifTrue, , -1)  ' Skip loop
+        ElseIf optConsume("while") Then
+            Call parseExpression                                     ' Break Condition
+            RepeatLoopJumpIndex = addOperation(oJump, ifFalse, , -1) ' Skip loop
+        Else
+            ' Infinite loop as long as no exit do is used
+        End If
+        Call parseBlock("exit", "loop")
+        If optConsume("exit") Then
+            ExitLoopJumpIndex = addOperation(oJump, , , -1) ' Skip loop
+            Call parseBlock("loop")
+        End If
+        Call addOperation(oJump, , RepeatLoopIndex, -1)
+        Call consume("loop")

```

Comment:
Consider adding an opt consume at this stage for:

```
do
  ...
loop while {expr}
```

or

```vb
do
  ...
loop until {expr}
```

I know this syntax exists in VBA, but I don't really know if it is valuable.